File: service/db/src/main/java/org/delivery/db/user/UserRepository.java
/*
package org.delivery.db.user;

import java.util.Optional;

import org.delivery.db.user.vo.UserStatus;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserRepository extends JpaRepository<User, Long> {

	Optional<User> findFirstByIdAndStatusOrderByIdDesc(Long userId, UserStatus status);
	Optional<User> findFirstByEmailAndPasswordAndStatusOrderByIdDesc(String email, String password, UserStatus status);

}
*/

File: delivery/service/db/src/main/kotlin/org/delivery/db/user/UserRepository.kt
package org.delivery.db.user

import org.delivery.db.user.vo.UserStatus
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<User, Long> {

    fun findFirstByIdAndStatusOrderByIdDesc(userId: Long?, status: UserStatus?): User?
    fun findFirstByEmailAndPasswordAndStatusOrderByIdDesc(
        email: String?,
        password: String?,
        status: UserStatus?
    ): User?

}